database/migrations/20191217071605_create-recipes-table.js

```diff
@@ -0,0 +1,50 @@
+
+exports.up = async function(knex) {
+    await knex.schema.createTable('recipes', tbl => {
+        tbl
+            .increments();
+        tbl
+            .text('title')
+            .notNullable();
+        tbl
+            .text('description')
+            .notNullable();
+        tbl
+            .text('instructions')
+            .notNullable();
+        tbl
+            .specificType('time_required', 'SMALLINT')
+            .notNullable();
+        tbl
+            .integer('difficulty')
+            .notNullable()
+            .unsigned();
+        tbl
+            .integer('budget')
+            .notNullable()
+            .unsigned();
+        tbl
+            .integer('user_id')
+            .unsigned()
+            .notNullable()
+            .references('id')
+            .inTable('users')
```

Comment:
Ditto. And elsewhere. :))

Reply:
Understood. I see your reasoning. I declare this conflict resolved